### PR TITLE
Update headers attribute refer to data cells ensuring ancestor table has a role of table grid or treegrid

### LINKED ISSUE #1928: Headers attribute rule should skip non-table roles [a25f45]
The following example would fail the [headers attribute rule](https://www.w3.org/WAI/standards-guidelines/act/rules/a25f45/), because the applicability only excludes table that are not in the accessibility tree. I think the exception should be broader, and exclude tables that do not have role=table, grid, or treegrid.

```html
<table role="region">
  <td id="self" headers="self">World</td>
</table>
```

### PR DESCRIPTION
**Pull request by giacomo-petri**
This pull request addresses the issue raised by @WilcoFiers, ensuring that headers attributes specified on a cell refers to cells in the same table element, when the table element has a role of table, grid or treegrid:
- added specific point about role in applicability;
- added specific inapplicable example.

Closes issue(s):

- closes #1928

Need for Call for Review:
This will require a 2 week Call for Review

---

### **When creating PR:**

- [ ] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [ ] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [ ] Add yourself (and co-authors) as "Assignees" for PR.
- [ ] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [ ] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [ ] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Comment by carlosapaduarte** [comment]
@giacomo-petri we're updating the applicability of the rule, therefore the call for review should be for 2 weeks

File: _rules/table-headers-attribute-refer-to-data-cells-a25f45.md [hunk 1]
```diff
@@ -28,7 +28,10 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any `headers` attribute specified on a [`cell`][] within a [`table`][] element, where the [`table`][] element is [visible][] and [included in the accessibility tree][].
+This rule applies to any `headers` attribute specified on a [`cell`][] within a [`table`][] element, where the [`table`][] element:
+- is [visible][]; and
+- is [included in the accessibility tree][]; and
+- has a role of `table`, `grid` or `treegrid`.
```
Comment (WilcoFiers) [inline _rules/table-headers-attribute-refer-to-data-cells-a25f45.md]:
```suggestion
This rule applies to any `headers` attribute specified on a [`cell`][] within a [`table`][] element, where all of the following is true for the [`table`][] element:

- The `table` is [visible][]; and
- The `table` is [included in the accessibility tree][]; and
- The `table` has a [semantic role][] of `table`, `grid` or `treegrid`.
```

**Review by WilcoFiers — DISMISSED** [review]
Couple small formatting tweaks. Rest looks good.

File: _rules/table-headers-attribute-refer-to-data-cells-a25f45.md [hunk 2]
```diff
@@ -395,6 +399,16 @@ The `table` is not [included in the accessibility tree][].
 </table>
 ```
 
+#### Inapplicable Example 6
+
+The `table` doesn't have a role of `table`, `grid` or `treegrid`.
```
Comment (Jym77) [inline _rules/table-headers-attribute-refer-to-data-cells-a25f45.md]:
```suggestion
This `table` doesn't have a role of `table`, `grid` or `treegrid`.
```

**Review by Jym77 — APPROVED** [review]

**Comment by Jym77** [comment]
(test fails due to the missing `[semantic role]: #semantic-role 'Definition of semantic role'` in the references list)

**Review by HelenBurge — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

**Review by WilcoFiers — APPROVED** [review]

**Review by colabottles — APPROVED** [review]

**Pull request merged**